src/coreComponents/linearAlgebra/interfaces/hypre/HypreInterface.cpp

```diff
@@ -18,16 +18,27 @@
 
 #include "HypreInterface.hpp"
 #include "linearAlgebra/interfaces/hypre/HyprePreconditioner.hpp"
+#include "HYPRE_utilities.h"
+#include "_hypre_utilities.h"
 
 namespace geosx
 {
 
 void HypreInterface::initialize( int & GEOSX_UNUSED_PARAM( argc ),
                                  char * * & GEOSX_UNUSED_PARAM( argv ) )
-{}
+{
+  HYPRE_Init();
+#if defined(USE_CUDA)
+  HYPRE_ExecutionPolicy default_exec_policy = HYPRE_EXEC_DEVICE;
+  hypre_HandleDefaultExecPolicy(hypre_handle()) = HYPRE_EXEC_DEVICE;
+  hypre_HandleSpgemmUseCusparse(hypre_handle()) = 1;
+#endif
+}
 
 void HypreInterface::finalize()
-{}
+{
+  HYPRE_Finalize();
+}
```

Comment:
@oseikuffuor1 @mquanbui This is the setup. I didn't know if this had to be called at the start of program execution or a later date....see below...

Reply:
@rrsettgast This is fine. See comment above and response to your comment below.